compiler/modules/types/xml.bal

```diff
@@ -23,46 +23,39 @@ const int XML_PRIMITIVE_RO_SINGLETON = XML_PRIMITIVE_TEXT | XML_PRIMITIVE_ELEMEN
 const int XML_PRIMITIVE_RO_MASK = XML_PRIMITIVE_NEVER | XML_PRIMITIVE_RO_SINGLETON;
 const int XML_PRIMITIVE_RW_MASK = XML_PRIMITIVE_ELEMENT_RW | XML_PRIMITIVE_PI_RW | XML_PRIMITIVE_COMMENT_RW;
 const int XML_PRIMITIVE_SINGLETON = XML_PRIMITIVE_RO_SINGLETON | XML_PRIMITIVE_RW_MASK;
+const int XML_PRIMITIVE_ALL_MASK = XML_PRIMITIVE_RO_MASK | XML_PRIMITIVE_RW_MASK;
 
-final XmlSubtype xmlRoTop = { primitives: XML_PRIMITIVE_RO_MASK, sequence: bddAtom(XML_PRIMITIVE_RO_SINGLETON) };
-final XmlSubtype xmlRwTop = { primitives: XML_PRIMITIVE_RW_MASK, sequence: bddAtom(XML_PRIMITIVE_SINGLETON) };
+final XmlSubtype xmlRoSubtype = { primitives: XML_PRIMITIVE_RO_MASK, sequence: bddAtom(XML_PRIMITIVE_RO_SINGLETON) };
+final XmlSubtype xmlTopSubtype = { primitives: XML_PRIMITIVE_ALL_MASK, sequence: true };
 
 function xmlSingleton(int primitives) returns SemType {
-    return createXmlSemtype(
-        createXmlSubtype(true, primitives, false), 
-        createXmlSubtype(false, primitives, false)
-    );
+    return createXmlSemType(createXmlSubtype(primitives, false));
 }
 
```

Comment:
Good to document that it is a precondition that constituentType is a subtype of XML. Good for robustness to panic if not.